### PR TITLE
Re-merge #187, New Missions PR

### PR DESCRIPTION
**Pull request by wersal454**
## What type of PR is this?
1. [ ] Bug
2. [x] Change
3. [x] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:
https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate/commit/baca9c2135e7cfb91c12ade7d5a54f6ae63b4795#:~:text=Insights-,Commit%20baca9c2,-Beta

Contains new missions pr #187

And maybe something else

### Please specify which Issue this PR Resolves (If Applicable).
"This PR closes #XXXX!"

### Please verify the following.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [ ] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

********************************************************
Notes:

### CODE REVIEW

File: A3A/addons/core/functions/Base/fn_destroyCity.sqf
Comment (Silence-112) [inline A3A/addons/core/functions/Base/fn_destroyCity.sqf:1]: the _markerX param was removed and _buildings is duplicated here

File: A3A/addons/core/functions/Base/fn_destroyCity.sqf [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+private _buildings = _positionX nearobjects ["house",_size];
 private _buildings = _positionX nearObjects ["house",_size];
 
 {
+	if (random 100 < 70) then {
+		for "_i" from 1 to 7 do {
+			_x setHit [format ["dam%1",_i],1];
+			_x setHit [format ["dam %1",_i],1];
+		};
+	};
```
Comment (Silence-112) [inline A3A/addons/core/functions/Base/fn_destroyCity.sqf]: Why is this necessary? Spamming setHit across the network isn't a great idea

File: A3A/addons/core/functions/Base/fn_mrkWIN.sqf
Comment (Silence-112) [inline A3A/addons/core/functions/Base/fn_mrkWIN.sqf:1]: This is from #296, it was only ever pushed to the testing branch. It shouldn't be pushed to unstable yet

File: A3A/addons/core/functions/CREATE/fn_AIVEHinit.sqf
Comment (Silence-112) [inline A3A/addons/core/functions/CREATE/fn_AIVEHinit.sqf:1]: Remove commented code if it's not needed, or add a comment explaining why it needs to stay

File: A3A/addons/core/functions/CREATE/fn_createAIAirplane.sqf
Comment (Silence-112) [inline A3A/addons/core/functions/CREATE/fn_createAIAirplane.sqf:1]: This is from #296, it was only ever pushed to the testing branch. It shouldn't be pushed to unstable yet

File: A3A/addons/core/functions/Missions/fn_CON_Outpost_Compet.sqf
Comment (Silence-112) [inline A3A/addons/core/functions/Missions/fn_CON_Outpost_Compet.sqf:1]: Remove commented code if it's not needed, or add a comment explaining why it needs to stay

File: A3A/addons/core/functions/Missions/fn_LOG_Salvage.sqf [hunk 2]
```diff
@@ -84,6 +85,45 @@ private _typeVeh = if (_difficultX) then { selectRandom (_faction get "vehiclesG
...
+			[_diver] call A3A_fnc_NATOinit;
+   		};
+   };
+	private _diversWaypoint = _diversGroup addWaypoint [selectRandom _markers, 30];
+	_diversWaypoint setWaypointType "SAD";
+	_diversGroup setCurrentWaypoint _diversWaypoint;
+	private _divers2Waypoint = _diversGroup2 addWaypoint [selectRandom _markers, 30];
+	_divers2Waypoint setWaypointType "SAD";
+	_diversGroup2 setCurrentWaypoint _divers2Waypoint;
+};
```
Comment (Silence-112) [inline A3A/addons/core/functions/Missions/fn_LOG_Salvage.sqf]: We shouldn't necessarily assume that templates with the "vanilla" flag also have vehiclesSDV defined

File: A3A/addons/core/functions/Missions/fn_LOG_Salvage.sqf [hunk 3]
```diff
@@ -147,3 +187,8 @@ deleteVehicle _ship;
 
 [_vehCrewGroup] spawn A3A_fnc_groupDespawner;
 [_veh] spawn A3A_fnc_vehDespawner;
+if ("vanilla" in A3A_factionEquipFlags) then {
+	[_SDVcrewGroup] spawn A3A_fnc_groupDespawner;
+	[_diversGroup] spawn A3A_fnc_groupDespawner;
+	[_veh] spawn A3A_fnc_vehDespawner;
+};
```
Comment (Silence-112) [inline A3A/addons/core/functions/Missions/fn_LOG_Salvage.sqf]: We also shouldn't assume that these groups were ever made just because the "vanilla" flag is set

Also is the `[_veh] spawn A3A_fnc_vehDespawner;` duplication necessary?

File: A3A/addons/core/functions/Missions/fn_missionRequest.sqf [hunk 4]
```diff
@@ -285,11 +288,20 @@ switch (_type) do {
...
+								[[_site],"A3A_fnc_RES_Prisoners"] remoteExec ["A3A_fnc_scheduler",2];
+							};
+						} else {
+							[[_site],"A3A_fnc_RES_Prisoners"] remoteExec ["A3A_fnc_scheduler",2];
+						};
 					};
 				};
 			};
-			if (_site in citiesX) then {[[_site],"A3A_fnc_RES_Refugees"] remoteExec ["A3A_fnc_scheduler",2]} else {[[_site],"A3A_fnc_RES_Prisoners"] remoteExec ["A3A_fnc_scheduler",2]};
+			//if (_site in citiesX) then {[[_site],"A3A_fnc_RES_Refugees"] remoteExec ["A3A_fnc_scheduler",2]} else {[[_site],"A3A_fnc_RES_Prisoners"] remoteExec ["A3A_fnc_scheduler",2]};
```
Comment (Silence-112) [inline A3A/addons/core/functions/Missions/fn_missionRequest.sqf]: This commented segment should be explained

File: A3A/addons/scrt/UI/fn_ui_assignRivalsAttackLocationEventHandler.sqf
Comment (Silence-112) [inline A3A/addons/scrt/UI/fn_ui_assignRivalsAttackLocationEventHandler.sqf:1]: This is a bit of a mess to read, it should be cleaned up a bit

**Review by Silence-112 — CHANGES_REQUESTED** [review]
Can't [fn_reDressInv](https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate/blob/c56325d5cec7d8b1aabc284b02b4adb837920153/A3A/addons/core/functions/REINF/fn_reDressInv.sqf) and [fn_reDressOcc](https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate/blob/c56325d5cec7d8b1aabc284b02b4adb837920153/A3A/addons/core/functions/REINF/fn_reDressOcc.sqf) be combined into one function by providing an extra `_side` parameter?

File: A3A/addons/core/functions/Base/fn_destroyCity.sqf (hunk 1 above)
Comment (wersal454) [reply A3A/addons/core/functions/Base/fn_destroyCity.sqf]: oh, this one isn't mine actually, don't know where it came from

Comment (wersal454) [reply A3A/addons/core/functions/Base/fn_destroyCity.sqf]: or maybe it is and it's not from #187

**Review by Silence-112 — COMMENTED** [review]
Seems fine, will need to be tested again

File: A3A/addons/core/functions/Base/fn_destroyCity.sqf (hunk 1 above)
Comment (Silence-112) [reply A3A/addons/core/functions/Base/fn_destroyCity.sqf]: Might aswell just remove this to be honest

File: A3A/addons/core/functions/REINF/fn_reDressFaction.sqf
Comment (Silence-112) [inline A3A/addons/core/functions/REINF/fn_reDressFaction.sqf:1]: This function doesn't get called anywhere. It's fine otherwise
